### PR TITLE
Rpaasv2 certificate command implemented

### PR DESCRIPTION
**Pull request by arthurcgc**

### CODE REVIEW

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		}
+
+		return runCert(certInst)
+	},
+}
+
+func encodeBody(certInst certificateArgs) (string, string, error) {
+	certBytes, err := ioutil.ReadFile(certInst.certificate)
+	if err != nil {
+		return "", "", err
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: I think we can wrap these errors to be more readable to the user.

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	keyFile, err := ioutil.ReadFile(certInst.key)
+	if err != nil {
+		return "", "", err
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: Same here...

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	certPart, err := writer.CreateFormFile("cert", certInst.certificate)
+	if err != nil {
+		return "", "", err
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: here...

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	_, err = certPart.Write(certBytes)
+	if err != nil {
+		return "", "", err
+	}
+
+	keyPart, err := writer.CreateFormFile("key", certInst.key)
+	if err != nil {
+		return "", "", err
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: here...

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	}
+	_, err = keyPart.Write(keyFile)
+	if err != nil {
+		return "", "", err
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: here...

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	err = writer.Close()
+	if err != nil {
+		return "", "", err
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: and here.

File: cmd/plugin/rpaasv2/cmd/status.go [hunk 7]
```diff
@@ -35,8 +35,6 @@ var statusCmd = &cobra.Command{
 		status.service = cmd.Flag("service").Value.String()
 		status.instance = cmd.Flag("instance").Value.String()
 		status.prox = proxy.New(status.service, status.instance, "GET", &proxy.TsuruServer{})
-		// status.prox = &proxy.Proxy{ServiceName: status.service, InstanceName: status.instance, Method: "GET"}
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/status.go]: You can safely drop these.

File: cmd/plugin/rpaasv2/cmd/certificate.go [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/tsuru/rpaas-operator/cmd/plugin/rpaasv2/proxy"
+)
+
+func init() {
+	rootCmd.AddCommand(certificateCmd)
+
+	certificateCmd.Flags().StringP("service", "s", "", "Service name")
+	certificateCmd.Flags().StringP("instance", "i", "", "Service instance name")
+	certificateCmd.Flags().StringP("certificate", "c", "", "Certificate file name")
+	certificateCmd.Flags().StringP("key", "k", "", "Key file name")
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/cmd/certificate.go]: On RPaaS v2 web API, we're accepting a new field `name` which gives a name to the provided certificate-key, it defaults to `default` value when not sent. Can you add this?

File: cmd/plugin/rpaasv2/go.mod [hunk 9]
```diff
@@ -3,10 +3,16 @@ module github.com/tsuru/rpaas-operator/cmd/plugin/rpaasv2
 go 1.12
 
 require (
+	github.com/docker/docker v0.7.3-0.20190327010347-be7ac8be2ae0 // indirect
```
Comment (nettoclaudio) [inline cmd/plugin/rpaasv2/go.mod]: Do these changes really necessary?

Comment (arthurcgc) [reply cmd/plugin/rpaasv2/go.mod]: I have no idea... Go modules...

File: cmd/plugin/rpaasv2/cmd/certificate.go (hunk 8 above)
Comment (arthurcgc) [reply cmd/plugin/rpaasv2/cmd/certificate.go]: On it.

File: cmd/plugin/rpaasv2/go.mod (hunk 9 above)
Comment (nettoclaudio) [reply cmd/plugin/rpaasv2/go.mod]: rofl

File: cmd/plugin/rpaasv2/cmd/status.go (hunk 7 above)
Comment (arthurcgc) [reply cmd/plugin/rpaasv2/cmd/status.go]: weird, this comment does not show on my local branch and when i run git diff on the remote it doesn't show the difference either...

**Pull request merged**